ct-app/core/components/utils.py

```diff
@@ -23,31 +23,40 @@ def nodesCredentials(
 
     @classmethod
     async def mergeDataSources(
-        cls,
-        topology: list,
-        peers: list,
-        safes: list,
+        cls, topology: list, peers: list, nodes: list, allocations: list
     ):
         for peer in peers:
-            address = peer.address.address
-            topo = next(filter(lambda t: t.node_address == address, topology), None)
-            safe = next(filter(lambda s: s.node_address == address, safes), None)
+            address = peer.node_address
+            topo = next(filter(lambda t: t.address == address, topology), None)
+            node = next(filter(lambda n: n.address == address, nodes), None)
 
-            if safe is None:
-                safe = NodeSafeEntry(address, "0", "0x0", "0")
+            peer.safe = getattr(node, "safe", SafeEntry.default())
 
-            if topo is not None and safe is not None and peer is not None:
+            for allocation in allocations:
+                if peer.safe.address in allocation.linked_safes:
+                    peer.safe.additional_balance += (
+                        allocation.unclaimed_amount / allocation.num_linked_safes
```

Comment:
Here it only adds the unclaimed allocation on top of their safe balances. Actually, the "claimed" part should NOT be neglected. However, to avoid double counting, it's better to just take the "wxHOPR balance of the recipient" into the equation before dividing by the "num_linked_safes"